### ISSUE TITLE
No warning returned for trying to import scikit-image when in source directory without building

### ISSUE THREAD

**Issue by michaelpacer** [scikit-image/scikit-image#1601]
E.g., if python initialized within the source directory and you try 

    from skimage.draw import ellipse

The following is lowest level of the returned error

    1 # coding: utf-8
    2 import numpy as np
    ----> 3 from ._draw import _coords_inside_image

    ImportError: No module named 'skimage.draw._draw'

this is despite the _draw.pyx file existing.

You would have needed to compile the cython to get it to work first.

**Comment by jjhelmus** [comment]
scikit-learn tests for a stub C extension and offers a more user friendly user error when importing from an unbuilt source directory.  This functionality is contained in the [sklean/__check_build directory](https://github.com/scikit-learn/scikit-learn/tree/master/sklearn/__check_build).  I'll make a PR for this shortly.

**Comment by michaelpacer** [comment]
Hi @jjhelmus just pushed my progress to https://github.com/michaelpacer/scikit-image/tree/check_build_sprint, which you should be able to fetch directly if you figure it out before I do (not sure when I'll be able to look at it next…but I think coming at it later with a fresh mind will help me).

**Comment by jni** [comment]
@jjhelmus @michaelpacer this would be a tremendous contribution, thanks for checking it out! We get so many support requests that would be fixed by a nicer error message here!

**Comment by jjhelmus** [comment]
@michaelpacer I added the the necessary logic to build the *_check_build* module from the master setup.py in my [check_build_sprint branch](https://github.com/jjhelmus/scikit-image/tree/check_build_sprint) which should be ready for a pull request.  I can submit this if you would like.  Alternatively you can grab the two extra commits from that branch (or I could submit a PR against your branch) and then you can submit the pull request.  Let me know which method you prefer.

**Comment by jjhelmus** [comment]
Example of the new build checking machinery.

Prior to build:
```
 ~/dev/scikit-image$ python -c "import skimage"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "skimage/__init__.py", line 125, in <module>
    from . import __check_build
  File "skimage/__check_build/__init__.py", line 47, in <module>
    raise_build_error(e)
  File "skimage/__check_build/__init__.py", line 42, in raise_build_error
    %s""" % (e, local_dir, ''.join(dir_content).strip(), msg))
ImportError: No module named _check_build
___________________________________________________________________________
Contents of skimage/__check_build:
README                    __init__.pyc              __init__.py
_check_build.pyx          setup.py
___________________________________________________________________________
It seems that scikit-image has not been built correctly.

If you have installed scikit-image from source, please do not forget
to build the package before using it: run `python setup.py install` or
`make` in the source directory.

It appears that you are importing a local scikit-image source tree. For
this, you need to have an inplace install. Maybe you are in the source
directory and you need to try from another location.
```

After build:
``` Shell
~/dev/scikit-image$ python setup.py build_ext -i
Partial import of skimage during the build process.
...
~/dev/scikit-image$ python -c "import skimage"
~/dev/scikit-image$
```

**Comment by jjhelmus** [comment]
FYI I submitted a pull request for this branch in #1622

**Comment by jjhelmus** [comment]
This issue can be closed as the merged PR #1622 address the problem.

**Comment by JDWarner** [comment]
Closed by #1622

**Comment by michaelpacer** [comment]
@jjhelmus @JDWarner Apologies for not noticing this sooner — I'd been away from all this git for a couple weeks, but I'm happy that this got completed and merged.  Thank you a tonne for this and https://github.com/scikit-image/scikit-image/pull/1622!
